### PR TITLE
Fix issue where cookie banner couldn't be accepted from privacy page

### PR DESCRIPTION
**Pull request by callumacrae**
### Context

https://dfedigital.atlassian.net/browse/CPDRP-407

### Guidance to review

To replicate the original issue:

- Go to e.g. https://ecf-review-pr-493.london.cloudapps.digital/
- DON'T TOUCH the cookies
- log in as school-leader@example.com
- On priv policy page, try to accept cookies

### Testing

Unsure if this needs request specs. I can't see how this would change in the future without invalidating the tests.

### How can I view this in a review app?

see pr app below

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -50,6 +50,9 @@ def check_privacy_policy_accepted
     policy = PrivacyPolicy.current
     return if policy.nil?
 
+    return if request.format == "application/json"
+    return if controller_name == "cookies"
```
Comment (kicferk1) [inline app/controllers/application_controller.rb:54]: That feels a bit dirty.

Can we have a protected `requires_privacy_policy` method on this controller, have it return `true` by default and then in cookies controller we override that method to return `false`? 

And in this place we `return if requires_privacy_policy`?

Comment (callumacrae) [reply app/controllers/application_controller.rb:54]: good shout 👍

Comment (callumacrae) [reply app/controllers/application_controller.rb]: found an even nicer solution!

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -50,6 +50,8 @@ def check_privacy_policy_accepted
     policy = PrivacyPolicy.current
     return if policy.nil?
 
+    return if request.format == "application/json"
```
Comment (callumacrae) [inline app/controllers/application_controller.rb:53]: this is technically unrelated to this PR, but it feels like it's probably a good idea

File: app/controllers/cookies_controller.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class CookiesController < ApplicationController
+  skip_before_action :check_privacy_policy_accepted
+
   before_action :set_cookie_form, only: :show
```
Comment (kicferk1) [inline app/controllers/cookies_controller.rb:6]: oh yeah, it is nicer <3

**Review by kicferk1 — APPROVED** [review]

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (ethanmills) [inline app/controllers/application_controller.rb:53]:
```suggestion
    return if request.format.json?
```

Does this work? We use a few mime types for JSON, and this should capture them all

**Review by ethanmills — APPROVED** [review]
One suggestion to try, don't forget to squash that squashable 🙂

**Pull request merged**